### PR TITLE
Make clonerefs setup commands create a record.

### PR DESCRIPTION
**Pull request by fejta**
/hold

### CODE REVIEW

**Review by cjwagner — COMMENTED** [review]
Thanks, this will make the clonerefs structured output and log a lot more complete.

File: prow/pod-utils/clone/types.go [hunk 1]
```diff
@@ -25,9 +25,9 @@ import (
 // what our final state ended up being, and
 // whether or not we were successful.
 type Record struct {
-	Refs     prowapi.Refs `json:"refs"`
-	Commands []Command    `json:"commands"`
-	Failed   bool         `json:"failed"`
+	Refs     prowapi.Refs `json:"refs,omitempty"`
```
[the remark this answers is not in the document]
Comment (fejta) [reply prow/pod-utils/clone/types.go:28]: Done

File: prow/clonerefs/run.go [hunk 2]
```diff
@@ -140,49 +171,89 @@ func needsGlobalCookiePath(cookieFile string, refs ...prowapi.Refs) string {
...
+		cmd.Error = err.Error()
+		cmds = append(cmds, cmd)
+		return "", cmds, fmt.Errorf("failed to close %s: %v", knownHostsFile, err)
 	}
+	cmds = append(cmds, cmd)
 	logrus.Infof("Updated known_hosts in file: %s", knownHostsFile)
 
 	ssh, err := exec.LookPath("ssh")
+	cmd = clone.Command{
+		Command: fmt.Sprintf("command -v ssh"),
```
[the remark this answers is not in the document]
Comment (fejta) [reply prow/clonerefs/run.go:230]: Done

File: prow/clonerefs/run.go [hunk 3]
```diff
@@ -98,16 +122,16 @@ func (o Options) Run() error {
 	wg.Wait()
 	close(output)
 
-	var hasFailedRecord bool
 	var results []clone.Record
```
[the remark this answers is not in the document]
Comment (fejta) [reply prow/clonerefs/run.go:125]: Done

**Review by cjwagner — APPROVED** [review]

**Pull request merged**